pkg/clustermesh/kvstoremesh/kvstoremesh.go

```diff
@@ -42,6 +45,9 @@ var DefaultConfig = Config{
 func (def Config) Flags(flags *pflag.FlagSet) {
 	flags.Duration("per-cluster-ready-timeout", def.PerClusterReadyTimeout, "Remote clusters will be disregarded for readiness checks if a connection cannot be established within this duration")
 	flags.Duration("global-ready-timeout", def.GlobalReadyTimeout, "KVStoreMesh will be considered ready even if any remote clusters have failed to synchronize within this duration")
+
+	flags.Bool("disable-drain-on-disconnection", def.DisableDrainOnDisconnection, "Do not drain cached data upon cluster disconnection")
```

Comment:
Please prefer `enable-drain-on-disconnection` rather than `disable-drain-on-connection`. Defining options as negative is confusing in some languages due to the double negative (if you _don't disable_ foo, ie `--disable-foo=false`, does that mean it's enabled or just that it's not explicitly enabled?)

Reply:
Alternative that may even be better: don't use `enable/disable` at all.

Reply:
I think we can still change the name of it - it is a hidden option as we did not expect anyone to use it, just as a fail-safe mechanism. 
But that's a really good point with naming, I was scratching my head a few times before with similar options too.

Reply:
> Please prefer enable-drain-on-disconnection rather than disable-drain-on-connection. Defining options as negative is confusing in some languages due to the double negative (if you don't disable foo, ie --disable-foo=false, does that mean it's enabled or just that it's not explicitly enabled?)

Hmm, I see your point. 

The reason for calling it `disable` is to avoid `true` by default boolean options, which also tend to be confusing and problematic in certain contexts (ie., a user may expect that entirely omitting `--enable-drain-on-disconnection=false` would be equivalent to specifying it, which would not be the case defaulting to true -- conversely --disable-drain-on-disconnection looks more intuitive to me). Although a bit unrelated, AFAIR a few true by default settings are also currently broken in the helm chart for similar reasons.

That said, as Marcel mentioned, this option is a temporary fail-safe (hidden) toggle not intended to be exposed to users, and I agree that we should still be able to change the name without particular consequences in case.

Reply:
>  [...] a user may expect that entirely omitting --enable-drain-on-disconnection=false would be equivalent to specifying it, which would not be the case defaulting to true [...]

To a degree I get the concern around default-true boolean options, but that's also just inherently how features work - features are either enabled or disabled. Depending on the maturity of the feature and how useful it is to the average user, we may decide to enable or disable different flags by default. There's then two important aspects to the UX - can users easily understand the default posture of the setting? And can they override it themselves?

Here's where I'm coming from in context of this specific flag: If I specify `--disable-drain-on-disconnection=false`, that means: Please don't "Do not drain cached data upon cluster disconnection".

This point is not a big deal for hidden options, but I raise it to encourage discussion about how we can provide more consistent UX across different flags. My gut feeling is that unless there is a really strong argument in a specific scenario, using "disable" (or other negative modifiers) in a flag name is almost always more confusing than the reverse.